test/testers.jl

```diff
@@ -42,6 +42,66 @@ primalapprox(x) = x
         end
     end
 
+    @testset "Inplace accumumulation: first on Array" begin
+        @testset "Correct definitions" begin
+            function ChainRulesCore.frule((_, ẋ), ::typeof(first), x::Array)
+                ẏ = InplaceableThunk(
+                    @thunk(first(ẋ)),
+                    ȧ -> ȧ + first(ẋ),  # This won't actually happen inplace
```

Comment:
Its hard to find forward-mode rules that are both simple and can be done in-place.
(first on a array of arrays is one of them i guess)

But i don't think we actually need something inplace to test this.

The real test of this functionality in detail is in the test for `check_result.jl`